master/pkg/syncx/mapx/mapx.go

```diff
@@ -0,0 +1,55 @@
+package mapx
+
+import "sync"
+
+// New creates an empty map.
+func New[K comparable, V any]() Map[K, V] {
+	return Map[K, V]{inner: make(map[K]V)}
+}
+
+// Map is a generic, thread-safe map to supersede usages of sync.Map.
+type Map[K comparable, V any] struct {
+	mu    sync.RWMutex
```

Comment:
> Using a raw mutex might cause linting issues on copy.

I think it is unwise to copy these any more-than-basic data structures with a standard shallow copy like `new := orig`. It doesn't make sense if the mutex is either copied or not copied, since the data is half copied and not copied depending on its type. So a raw mutex that causes a lint error is best.

Reply:
I don't think a data structure that facilitates a mutex per map is a huge deal; since the methods force you to acquire and release it without doing anything else (unless someone seriously misuses .WithLock), it is almost impossible to have a cycle-induced deadlock.